drafts/client-interface.md

```diff
@@ -3,11 +3,150 @@
 This document describes the interface of a yet-to-be-built reference client
 library in pseudo code.
 
+The hydra client works similar to a browser. You point it to a URL. The URL becomes the current resource. 
+Then you interact with the controls of the resource.
 
-## Retrieve a resource
+The examples show how a cli application could use the hydra client for interaction with an API.
+
+## Find API entrypoint for website
+
+Hydra has a special mechanism to discover API entrypoints.
+
+```
+hydra.findEntrypoint(website)
+print hydra.representation 
+// the API entrypoint is now the current representation
```

Comment:
It's probably only because I intended this to be a lower-level thing than what you proposed. I had envisioned the client library to be the thing that is being used by apps like the one you propose in this document. Could we perhaps document your Hydra Browser as a layer implemented on top of a more low-level Hydra Client? I'm completely fine with doing so in this file as I don't think we know yet which alternative is better